spec/controllers/api/v1/characters_controller_spec.rb

```diff
@@ -46,6 +63,26 @@
         expect(response.json['results']).to contain_exactly(char.as_json(include: [:selector_name]).stringify_keys)
       end
 
+      it "filters by template" do
+        template = create(:template)
+        char = create(:character, template: template)
+        create(:character)
+        get :index, params: { template_id: template.id }
+        expect(response).to have_http_status(200)
+        expect(response.json).to have_key('results')
+        expect(response.json['results']).to contain_exactly(char.as_json(include: [:selector_name]).stringify_keys)
+      end
+
+      it "filters by templateless" do
+        template = create(:template)
+        create(:character, template: template)
+        char = create(:character)
+        get :index, params: { template_id: '0' }
+        expect(response).to have_http_status(200)
+        expect(response.json).to have_key('results')
+        expect(response.json['results']).to contain_exactly(char.as_json(include: [:selector_name]).stringify_keys)
+      end
```

Comment:
Should we also test specific output for a specially constructed example? Like, create an icon and character and template with specific names and then make sure the output is a specific thing? (If we have good tests for the character presenter, this might not be necessary, but I notice we didn't add a test for `template_name` producing correct results.)

Reply:
Oh. I was confused by `template_name` and thinking it was literally the name of the template association. Never mind, since it's the template nickname and is definitely a property of character, I'm not concerned. (I am concerned by the fact we still have the confusing `character.template_name` in our code though XD)